contracts/Vault1.sol

```diff
@@ -1,6 +1,27 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
-contract Vault1 {
-    
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract Vault1 is ERC20 {
+    address public owner;
+    mapping(address => uint) balances;
+
+    constructor() ERC20('Vault1', 'V1'){
```

Comment:
This contract should support any ERC20 token - currently, it supports Vault1 token only. This contract need not be an ERC20, it can just take an ERC20 contract address in its constructor and then create an object of ERC20 and call functions on it